src/jaxsim/rbda/contacts/rigid.py

```diff
@@ -0,0 +1,577 @@
+    ) -> RigidContactParams:
+        """Create a `RigidContactParams` instance"""
+        return RigidContactParams(
+            mu=mu, K=K, D=D, inactive_points_prev=inactive_points_prev
+        )
+
+    @staticmethod
+    def build_from_jaxsim_model(
+        model: js.model.JaxSimModel,
+        *,
+        static_friction_coefficient: jtp.Float = 0.5,
+        K: jtp.Float = 0.0,
+        D: jtp.Float = 0.0,
+    ) -> RigidContactParams:
+        """Build a `RigidContactParams` instance from a `JaxSimModel`."""
+
+        inactive_points_prev = jnp.zeros(
+            model.kin_dyn_parameters.contact_parameters.point.shape[0]
+        )
+
+        jax.debug.print(
+            "==========inactive_points_prev={inactive_points_prev}",
+            inactive_points_prev=inactive_points_prev.shape,
+        )
+
+        return RigidContactParams.build(
+            mu=static_friction_coefficient,
+            K=K,
+            D=D,
+            inactive_points_prev=inactive_points_prev,
+        )
+
+    def valid(self) -> bool:
+        return bool(
+            jnp.all(self.mu >= 0.0)
+            and jnp.all(self.K >= 0.0)
+            and jnp.all(self.D >= 0.0)
+        )
+
+
+@jax_dataclasses.pytree_dataclass
+class RigidContactsState(ContactsState):
+    """Class storing the state of the rigid contacts model."""
+
+    def __hash__(self) -> int:
+        return hash(tuple(jnp.atleast_1d(self.inactive_points_prev.flatten()).tolist()))
+
+    def __eq__(self, other: RigidContactsState) -> bool:
+        return hash(self) == hash(other)
+
+    @staticmethod
+    def build_from_jaxsim_model(
+        model: js.model.JaxSimModel | None = None,
+    ) -> RigidContactsState:
+        """Build a `RigidContactsState` instance from a `JaxSimModel`."""
+        return RigidContactsState.build()
+
+    @staticmethod
+    def build() -> RigidContactsState:
+        """Create a `RigidContactsState` instance"""
+        return RigidContactsState()
+
+    @staticmethod
+    def zero(model: js.model.JaxSimModel) -> RigidContactsState:
+        """Build a zero `RigidContactsState` instance from a `JaxSimModel`."""
+        return RigidContactsState.build()
+
+    def valid(self, model: js.model.JaxSimModel) -> bool:
+        return True
+
+
+@jax_dataclasses.pytree_dataclass
+class RigidContacts(ContactModel):
+    """Rigid contacts model."""
+
+    parameters: RigidContactParams = dataclasses.field(
+        default_factory=RigidContactParams
+    )
+
+    terrain: jax_dataclasses.Static[Terrain] = dataclasses.field(
+        default_factory=FlatTerrain
+    )
+
+    @staticmethod
+    def _detect_contacts(
+        W_o_C: jtp.Array,
+        W_o_dot_C: jtp.Array,
+        terrain_height: jtp.Array,
+        terrain_normal: jtp.Array,
+    ) -> tuple[jtp.Vector, tuple[Any, ...]]:
+        """
+        Detect contacts between the collidable points and the terrain.
+        """
+
+        # TODO: reduce code duplication with js.contact.in_contact
```

Comment:
This is a good idea! 

The concept behind `jaxsim.api` is to provide some "high level" functions, so this should be probably put in `jaxsim.rbda.contacts.common`. Then `js.contact.in_contact` will call `jaxsim.rbda.contacts.common.detect_contacts`. At that point, it could be applied also in `jaxsim.rbda.contacts.soft`

Reply:
Yes we could uniform the contact detection mechanism and put it into `jaxsim.rbda.contacts.common`